pkg/tkctl/cmd/upinfo/upinfo.go

```diff
@@ -183,25 +216,21 @@ func renderTCUpgradeInfo(tc *v1alpha1.TidbCluster, set *apps.StatefulSet, podLis
 							return err
 						}
 						if dbPhase == v1alpha1.UpgradePhase {
-							if (*updateReplicas) < ordinal {
-								state = UPDATED
-							} else if (*updateReplicas) == ordinal {
-
-								state = UPDATING
-
-								if pod.Labels[apps.ControllerRevisionHashLabelKey] == tc.Status.TiDB.StatefulSet.UpdateRevision {
-									if member, exist := tc.Status.TiDB.Members[pod.Name]; exist && member.Health {
-										state = UPDATED
-									}
-								}
-
-							} else {
-								state = WAITING
-							}
+							state = getState(*updateReplicas, ordinal, tc, &pod)
 						} else {
 							state = UPDATED
 						}
-						w.WriteLine(readable.LEVEL_1, "%s\t%s\t", pod.Name, state)
+
+						container, ok := util.GetContainerViaName(v1alpha1.TiDBMemberType.String(), &pod)
+						if !ok {
+							return errors.New("no tidb container")
```

Comment:
As of a CLI command, it's better to render `<none>` or error message instead of return an error. This can guarantee that other informations are  still rendered and won't be hidden by this error.

Reply:
make sense